base_multi_image/models/image.py

```diff
@@ -0,0 +1,222 @@
+        "Medium-sized image",
+        compute="_get_image_sizes",
+        help="Medium-sized image. It is automatically resized as a "
+             "128 x 128 px image, with aspect ratio preserved, only when the "
+             "image exceeds one of those sizes. Use this field in form views "
+             "or kanban views.")
+    image_small = fields.Binary(
+        "Small-sized image",
+        compute="_get_image_sizes",
+        help="Small-sized image. It is automatically resized as a 64 x 64 px "
+             "image, with aspect ratio preserved. Use this field anywhere a "
+             "small image is required.")
+    comments = fields.Text(
+        'Comments',
+        translate=True)
+    sequence = fields.Integer(
+        default=10)
+    show_technical = fields.Boolean(
+        compute="_show_technical")
+
+    @api.model
+    @tools.ormcache("self")
+    def _selection_owner_ref_id(self):
+        """Allow any model; after all, this field is readonly."""
+        return [(r.model, r.name) for r in self.env["ir.model"].search([])]
+
+    @api.multi
+    @api.depends("owner_model", "owner_id")
+    def _compute_owner_ref_id(self):
+        """Get a reference field based on the split model and id fields."""
+        for s in self:
+            if s.owner_model:
+                s.owner_ref_id = "{0.owner_model},{0.owner_id}".format(s)
+
+    @api.multi
+    @api.depends('storage', 'path', 'file_db_store', 'url')
+    def _get_image(self):
+        """Get image data from the right storage type."""
+        for s in self:
+            s.image_main = getattr(s, "_get_image_from_%s" % s.storage)()
+
+    @api.multi
+    @api.depends("owner_id", "owner_model")
+    def _show_technical(self):
+        """Know if you need to show the technical fields."""
+        self.show_technical = all(
+            "default_owner_%s" % f not in self.env.context
+            for f in ("id", "model"))
+
+    @api.multi
+    def _get_image_from_filestore(self):
+        return self.attachment_id.datas
+
+    @api.multi
+    def _get_image_from_db(self):
+        return self.file_db_store
+
+    @api.multi
+    def _get_image_from_file(self):
+        if self.path and os.path.exists(self.path):
+            try:
+                with open(self.path, 'rb') as f:
+                    return base64.b64encode(f.read())
+            except Exception as e:
+                _logger.error("Can not open the image %s, error : %s",
+                              self.path, e, exc_info=True)
+        else:
+            _logger.error("The image %s doesn't exist ", self.path)
+
+        return False
+
+    @api.multi
+    def _get_image_from_url(self):
+        return self._get_image_from_url_cached(self.url)
+
+    @api.model
+    @tools.ormcache("url")
+    def _get_image_from_url_cached(self, url):
+        """Allow to download an image and cache it by its URL."""
+        if url:
+            try:
+                (filename, header) = urllib.urlretrieve(url)
```

Comment:
```suggestion
                (filename, header) = urlretrieve(url)
```
To go with previous comment.